simphony/material_relations/material_relation.py

```diff
@@ -0,0 +1,121 @@
+from simphony.core.data_container import DataContainer
+from simphony.core.material_relations_information import (
+    MATERIAL_RELATIONSHIP_INFORMATION)
+
+
+class MaterialRelation(object):
+    """ Class represents relationships between materials
+
+    MaterialRelation provides a general interface for describing the
+    (physics/chemistry) relations between different materials
+
+    Attributes
+    ----------
+    kind: CUDSMaterialRelation
+        Describes the kind of the MaterialRelation
+
+    name : str
+        name of the material-relation
+
+    materials: list of uids
+        materials where this relation applies
+
+    parameters: DataContainer
+        the required parameters
+
+    description: str
+        user-defined description of the material-relation
+
+    supported_parameters: list of CUBA
+        CUBA values required/allowed for the parameters
+
+    allowed_number_materials : list of int
+        list with all possible configurations of avaliable
+        number of materials in the relation
+
+    """
+
+    def __init__(self, name, kind, materials, parameters, description=""):
+
+        info = MATERIAL_RELATIONSHIP_INFORMATION[kind]
+        self._allowed_number_materials = info.allowed_number_materials
+        self._supported_parameters = [
+            parameter.cuba_key for parameter in info.supported_parameters]
+
+        self._name = name
+        self._kind = kind
+        self._description = description
+
+        self._check_materials(materials)
```

Comment:
you could also do `self.materials = materials` and `self.parameters = parameters` so you reuse the setters.